src/detect-engine-siggroup.c

```diff
@@ -630,7 +630,11 @@ void SigGroupHeadSetFilestoreCount(DetectEngineCtx *de_ctx, SigGroupHead *sgh)
             continue;
 
         if (SignatureIsFilestoring(s)) {
-            sgh->filestore_cnt++;
+            if (sgh->filestore_cnt == UINT16_MAX) {
+                SCLogWarning("Too many filestore in signature group head");
```

Comment:
Would this make `filestore` in sigs post the `UINT16_MAX` limit in `sgh` be ignored?
If yes, should we mention that too?

Reply:
Could we improve the message wording to be clearer on the cause of the issue so the user would have some guidance on what to do when this appears?

Reply:
Instead of a warning that will be hard to understand and act on, perhaps we just need a limit on the total number of filestore sigs loaded in a detect engine. We could track this in `DetectEngineCtx` during parsing, then error out each rule beyond 64k.

Reply:
Trying that in next PR